src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs

```diff
@@ -75,54 +76,91 @@ protected virtual IEnumerable<FieldType> ProvideFields()
+                fieldType.Arguments = null;
+                fieldType.Resolver = resolver;
             }
-
-            List<LambdaExpression> expressions = new();
-            QueryArguments queryArguments = new();
-            foreach (var parameterInfo in methodInfo.GetParameters())
+            else if (memberInfo is MethodInfo methodInfo)
             {
-                var argumentInfo = (ArgumentInformation?)_getArgumentInformationInternalMethodInfo
-                    .MakeGenericMethod(parameterInfo.ParameterType)
-                    .Invoke(this, new object[] { fieldType, parameterInfo })!;
-                var (queryArgument, expression) = argumentInfo.ConstructQueryArgument();
-                if (queryArgument != null)
+                List<LambdaExpression> expressions = new();
+                QueryArguments queryArguments = new();
+                foreach (var parameterInfo in methodInfo.GetParameters())
                 {
-                    ApplyArgumentAttributes(parameterInfo, queryArgument);
-                    queryArguments.Add(queryArgument);
+                    var getArgumentInfoMethodInfo = _getArgumentInformationInternalMethodInfo.MakeGenericMethod(parameterInfo.ParameterType);
+                    var getArgumentInfoMethod = (Func<FieldType, ParameterInfo, ArgumentInformation>)getArgumentInfoMethodInfo.CreateDelegate(typeof(Func<FieldType, ParameterInfo, ArgumentInformation>), this);
+                    var argumentInfo = getArgumentInfoMethod(fieldType, parameterInfo);
+                    var (queryArgument, expression) = argumentInfo.ConstructQueryArgument();
+                    if (queryArgument != null)
+                    {
+                        ApplyArgumentAttributes(parameterInfo, queryArgument);
+                        queryArguments.Add(queryArgument);
+                    }
+                    expression ??= AutoRegisteringHelper.GetParameterExpression(
+                        parameterInfo.ParameterType,
+                        queryArgument ?? throw new InvalidOperationException("Invalid response from ConstructQueryArgument: queryArgument and expression cannot both be null"));
+                    expressions.Add(expression);
                 }
-                expression ??= AutoRegisteringHelper.GetParameterExpression(
-                    parameterInfo.ParameterType,
-                    queryArgument ?? throw new InvalidOperationException("Invalid response from ConstructQueryArgument: queryArgument and expression cannot both be null"));
-                expressions.Add(expression);
+                var resolver = new MethodResolver(methodInfo, BuildMemberInstanceExpression(memberInfo), expressions);
+                fieldType.Arguments = queryArguments;
+                fieldType.Resolver = resolver;
+            }
+            else if (memberInfo is FieldInfo fieldInfo)
+            {
+                var sourceExpression = BuildMemberInstanceExpression(memberInfo);
```

Comment:
Property/Method branches handle static methods (for property it is static get method). What about fields?

Reply:
Good catch; fixed.  Test added as you noted above.

Reply:
👍